### PR TITLE
New Bear: PyCheckerBear

### PR DESCRIPTION
**Pull request by khanchi97**
A wrapper for `pychecker` which find bugs in python files.

Closes https://github.com/coala/coala-bears/issues/2151

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/python/PyCheckerBear.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        output_format='regex',
+        output_regex=r'(?P<filename>\w+\.py):(?P<line>\d+): '
+                     r'(?P<message>.*)')
+class PyCheckerBear:
+    """
+    Find bugs in your python source code.
+
+    The code for each function, class, and method is checked for possible
+    problems.Checks for unused globals and locals(module or variable),unused
+    method arguments and using a variable before setting it or if you are
```
Comment (manankalra) [inline bears/python/PyCheckerBear.py]: a single space after a comma and a full-stop looks way nicer ;)

Comment (manankalra) [inline bears/python/PyCheckerBear.py]: a single space after a comma and a full-stop looks way nicer ;)

**Comment by manankalra** [comment]
@khanchi97 I think we you use *double* back-ticks for a code-term in comments/docs. 
Also, which --> to **OR** find --> finds
python --> Python
Change your commit body accordingly.

File: bears/python/PyCheckerBear.py (hunk 1 above)
Comment (khanchi97) [reply bears/python/PyCheckerBear.py]: :+1:

**Comment by khanchi97** [comment]
Done :+1:

File: tests/python/PyCheckerBearTest.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        self.check_results(self.uut, load_testfiles('good_file.py'),
+                           [],
+                           get_testfile_path('good_file.py'))
+
+    def test_bad_file(self):
+        self.check_results(self.uut, load_testfiles('bad_file.py'),
+                           [Result.from_values('PyCheckerBear',
+                                               'Parameter (c) not used',
+                                               get_testfile_path(
+                                                   'bad_file.py'),
```
Comment (manankalra) [inline tests/python/PyCheckerBearTest.py]: You're using `get_testfile_path(...)` repeatedly. Maybe initialize a bad_file/good_file vars under `setUp()`.

File: tests/python/PyCheckerBearTest.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                                               line=1)],
+                           get_testfile_path('bad_file.py'))
```
Comment (manankalra) [inline tests/python/PyCheckerBearTest.py]: like here too

File: bears/python/PyCheckerBear.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        DistributionRequirement)
+
+
+@linter(executable='pychecker',
+        output_format='regex',
+        output_regex=r'(?P<filename>\w+\.py):(?P<line>\d+): '
+                     r'(?P<message>.*)')
+class PyCheckerBear:
+    """
+    Find bugs in your python source code.
```
Comment (manankalra) [inline bears/python/PyCheckerBear.py]: python --> Python ;)

File: bears/python/PyCheckerBear.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    problems. Checks for unused globals and locals(module or variable), unused
+    method arguments and using a variable before setting it or if you are
+    redefining a function/class/method in the same scope.
+    """
+    LANGUAGES = {'Python', 'Python 2'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='pychecker')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Unused Method/function arguments',
```
Comment (manankalra) [inline bears/python/PyCheckerBear.py]: Method --> method

File: bears/python/PyCheckerBear.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                  'No doc string', 'Redefining'}
+    SEE_MORE = 'http://pychecker.sourceforge.net/'
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         pychecker_cli_options: str=''):
+        """
+        :param pychecker_cli_options: Any command line options you wish to be
```
Comment (manankalra) [inline bears/python/PyCheckerBear.py]: *Any* can be removed

**Comment by khanchi97** [comment]
Done :+1:

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
Closing due to inactivity.

**Pull request closed without merging**